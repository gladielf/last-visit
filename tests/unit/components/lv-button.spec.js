import { shallowMount } from '@vue/test-utils';
import LvButton from '@/components/lv-button.vue';

describe('LvButton - Snapshot', () => {
    it('Snapshot', () => {
        const text = 'my button';
        const wrapper = shallowMount(LvButton, {
            propsData: { text }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});

describe('LvButton - methods', () => {
    it('btnClicked', async () => {
        const text = 'my button';
        const wrapper = shallowMount(LvButton, {
            propsData: { text }
        });

        wrapper.vm.btnClicked();

        await wrapper.vm.$nextTick();

        expect(wrapper.emitted('lv-button-clicked')).toBeTruthy();
    });
});
