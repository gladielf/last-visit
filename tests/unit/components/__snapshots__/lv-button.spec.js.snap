// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LvButton - Snapshot Snapshot 1`] = `
<button
  class="lv-button"
  type="button"
>
  my button
</button>
`;
